src/main/pages/che-7/hands-on-guidance/proc_choosing-the-sidecar-image.adoc

```diff
@@ -0,0 +1,48 @@
+// Module included in the following assemblies:
+//
+// using-a-visual-studio-code-extension-in-che
+
+// This module can be included from assemblies using the following include statement:
+// include::<path>/proc_choosing-the-sidecar-image.adoc[leveloffset=+1]
+
+// The file name and the ID are based on the module title. For example:
+// * file name: proc_doing-procedure-a.adoc
+// * ID: [id='proc_doing-procedure-a_{context}']
+// * Title: = Doing procedure A
+//
+// The ID is used as an anchor for linking to the module. Avoid changing
+// it after the module has been published to ensure existing links are not
+// broken.
+//
+// The `context` attribute enables module reuse. Every module's ID includes
+// {context}, which ensures that the module has a unique ID even if it is
+// reused multiple times in a guide.
+//
+// Start the title with a verb, such as Creating or Create. See also
+// _Wording of headings_ in _The IBM Style Guide_.
+[id="choosing-the-sidecar-image_{context}"]
+= Choosing the sidecar image
+
+Che plug-ins are special services that extend the Che workspace capabilities. Che plug-ins are packaged as containers. These containers that the plug-ins are packaged into,  run as sidecars of the Che workspace editor and augment its capabilities.
+
+[discrete]
+== Procedure
+
+To choose a sidecar image:
+
+. If the VS code extension does not have any external dependencies, use `eclipse/che-theia-endpoint-runtime: next` as a sidecar container image for the extension.
++
+*Note*: In addition to the base image `eclipse/che-theia-endpoint-runtime:next, the following ready-to-use sidecar images that include language specific dependencies are available:
+
+* eclipse/che-remote-plugin-runner-java8
+* eclipse/che-remote-plugin-runner-java11
+* eclipse/che-remote-plugin-go-1.10.7
+* eclipse/che-remote-plugin-python-3.7.3
+* eclipse/che-remote-plugin-dotnet-2.2.105
+* eclipse/che-remote-plugin-php7
+* eclipse/che-remote-plugin-kubernetes-tooling-1.0.0
+* eclipse/che-remote-plugin-kubernetes-tooling-0.1.17
+* eclipse/che-remote-plugin-openshift-connector-0.0.17
+* eclipse/che-remote-plugin-openshift-connector-0.0.21
++
+. If a VS Code extension has external dependencies that are not found in one of the images above, use a container image that contains the needed dependencies for the extension and is based on the `eclipse/che-theia-endpoint-runtime:next` image. Example: the `FROM` directive should look like `FROM eclipse/che-theia-endpoint-runtime:next`. This is required because this base image contains tools for running the remote VS Code extension and communications between the sidecar and the Che Theia editor, so that the VS Code extension does not have to know that it is a remote one.
```

Comment:
```suggestion
. If a VS Code extension has external dependencies that are not found in one of the preceding images, use a container image that contains the needed dependencies for the extension and is based on the `eclipse/che-theia-endpoint-runtime:next` image. Example: the `FROM` directive should look like `FROM eclipse/che-theia-endpoint-runtime:next`. This is required because this base image contains tools for running the remote VS Code extension and communications between the sidecar and the Che Theia editor so that the VS Code extension does not have to know that it is a remote one.
```